Backpack-SwiftUI/Calendar/Classes/Core/CalendarMonthGrid.swift

```diff
@@ -40,56 +40,74 @@ struct CalendarMonthGrid<
-                    emptyLeadingDayCell(emptyDayInfo)
-                        .frame(height: dayCellHeight)
-                    Spacer(minLength: BPKSpacing.none)
-                }
-            }
-
+            previousEmptyCells(daysFromPreviousMonth: daysFromPreviousMonth)
+            let numberOfDaysInMonth = calendar.range(of: .day, in: .month, for: monthDate)!.count
             // Create cells for the days in the current month
             currentMonthDayCell(numberOfDaysInMonth: numberOfDaysInMonth)
 
             // Create cells for the days from the next month that are shown in the last week of the current month
             // The total number of cells used is the sum of the number of days in the current month and the number of
             // days from the previous month that are shown
-            
-            if remainingCells < daysInAWeek {
-                ForEach(emptyDaysTrailing) { emptyDayInfo in
-                    VStack(spacing: BPKSpacing.none) {
-                        emptyTrailingDayCell(emptyDayInfo)
-                            .frame(height: dayCellHeight)
-                        Spacer(minLength: BPKSpacing.none)
-                    }
+            let totalCellsUsed = numberOfDaysInMonth + daysFromPreviousMonth
+            let remainingCells = daysInAWeek - (totalCellsUsed % daysInAWeek)
+        
+            remainingEmptyCells(remainingCells: remainingCells)
+        }
+    }
+    
+    @ViewBuilder
+    private func previousEmptyCells(daysFromPreviousMonth: Int) -> some View {
+        let preEmptyCells = Array(0..<daysFromPreviousMonth)
+            .map {
+                DayCellIdentifiable(id: "pre-\($0)\(monthDate)", index: $0)
+            }
+        ForEach(preEmptyCells) { _ in
+            VStack(spacing: BPKSpacing.none) {
+                emptyLeadingDayCell()
+                    .frame(height: dayCellHeight)
+                Spacer(minLength: BPKSpacing.none)
+            }
+        }
+    }
+    
+    @ViewBuilder
+    private func remainingEmptyCells(remainingCells: Int) -> some View {
+        if remainingCells < daysInAWeek {
+            let remainingEmptyCells = Array(0..<remainingCells)
+                .map {
+                    DayCellIdentifiable(id: "rem-\($0)\(monthDate)", index: $0)
+                }
+            ForEach(remainingEmptyCells) { _ in
+                VStack(spacing: BPKSpacing.none) {
+                    emptyTrailingDayCell()
+                        .frame(height: dayCellHeight)
+                    Spacer(minLength: BPKSpacing.none)
                 }
             }
         }
     }
     
+    private struct DayCellIdentifiable: Identifiable {
+        let id: String
+        let index: Int
+    }
+    
     @ViewBuilder
     private func currentMonthDayCell(numberOfDaysInMonth: Int) -> some View {
-        ForEach(0..<numberOfDaysInMonth, id: \.self) { cellIndex in
+        let days = Array(0..<numberOfDaysInMonth)
```

Comment:
same as above, using the struct rather than a ForEach with a range, that led to bugs before